src/nn/wasm/PreparedModel.js

```diff
@@ -1264,6 +1264,18 @@ export default class PreparedModel {
         OPS_CHECK(output.runtimeshape.DimensionsCount() <= 4);
         OPS_CHECK(output.runtimeshape.DimensionsCount() === perm.length);
 
+        if (perm.length < 4){
+          let extended_perm = [];
+
+          for (let i = 0; i < perm.length; ++i){
+            extended_perm[i] = perm[i];
+          }
+          for (let i = perm.length; i < 4; ++i){
+            extended_perm[i] = 0;
+          }
+          perm = extended_perm;
+        }
```

Comment:
can we just use `perm`?
```
// Extend perm to length 4 by appending 0
for (let i = perm.length; i < 4; ++i) {
  perm.push(0);
}
```

Reply:
@akineeic  any thoughts?

Reply:
Sorry I see your comment just now, it's better in this way. I'll modify it.

Reply:
no worries. looks good now.